owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/SettingsActivity.kt

```diff
@@ -58,12 +63,40 @@ class SettingsActivity : AppCompatActivity() {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             android.R.id.home -> {
-                intent = Intent(this, FileDisplayActivity::class.java).apply {
-                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                if (supportFragmentManager.findFragmentByTag("logs") != null) {
+                    supportFragmentManager.popBackStack()
+                    supportActionBar?.setTitle(R.string.actionbar_settings)
+                } else {
+                    intent = Intent(this, FileDisplayActivity::class.java).apply {
+                        addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                    }
+                    startActivity(intent)
                 }
-                startActivity(intent)
             }
         }
         return super.onOptionsItemSelected(item)
     }
+
+    override fun onPreferenceStartFragment(caller: PreferenceFragmentCompat, pref: Preference): Boolean {
+        val fragment = supportFragmentManager.fragmentFactory.instantiate(
+            classLoader,
+            pref.fragment)
+        var tag: String? = null
+        var titleId: Int? = null
+        if (pref.key.equals(SettingsFragment.PREFERENCE_LOGS_SUBSECTION)) {
+            tag = "logs"
+            titleId = R.string.actionbar_logger
+        }
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.settings_container, fragment, tag)
+            .addToBackStack(null)
+            .commit()
+        supportActionBar?.setTitle(titleId!!)
+        return true
+    }
+
+    override fun onBackPressed() {
+        super.onBackPressed()
+        supportActionBar?.setTitle(R.string.actionbar_settings)
+    }
```

Comment:
I would remove this if we add an `addOnBackStackChangedListener` and updating the toolbar in consecuence:

`        supportFragmentManager.addOnBackStackChangedListener {
            updateToolbarTitle()
        }`

And updateToolbarTitle() could be something like this:
```   
private fun updateToolbarTitle() {
        val titleId = when (supportFragmentManager.fragments.lastOrNull()) {
            is SettingsLogsFragment -> R.string.actionbar_logger
            else -> R.string.actionbar_settings
        }

        supportActionBar?.setTitle(titleId)
    }
```
And remember that if the device rotates, the activity is recreated and you won't need to add the SettingsFragment again because it will remember 👍 
```
        if (savedInstanceState != null) return

        supportFragmentManager
            .beginTransaction()
            .replace(R.id.settings_container, SettingsFragment())
            .commit()
```

🍻 🍻